VS2015 Version 14.0.25431.01 Update 3 compile error x64 RelWithDebInfo

Trying to build the repo in x64 RelWithDebInfo I get the following compile error in core_validation.cpp:

```
2>------ Build started: Project: VkLayer_core_validation, Configuration: RelWithDebInfo x64 ------
2>  core_validation.cpp
2>D:\Users\Dan\src\Vulkan-LoaderAndValidationLayers\layers\core_validation.cpp : fatal error C1128: number of sections exceeded object file format limit: compile with /bigobj
```

This can be worked around by adding /bigobj to the "Additional Options" for core_validation.cpp.  It would be convenient if this were done in cmake.  For example, here is a project that seems to do something like this for debug builds: (https://gitlab.com/trinitycore/TrinityCore_434/blob/2f8205e012b4717d4f68c97bf86d555f755f0d53/cmake/compiler/msvc/settings.cmake)

```
if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS 19.0.23026.0)
    #Enable extended object support for debug compiles on X64 (not required on X86)
    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /bigobj")
    message(STATUS "MSVC: Enabled increased number of sections in object files")
endif()
```

--- a/layers/CMakeLists.txt
+++ b/layers/CMakeLists.txt
@@ -174,12 +174,11 @@ include_directories(
 )
 
 if (WIN32)
-    set (CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -D_CRT_SECURE_NO_WARNINGS")
-    set (CMAKE_C_FLAGS_RELEASE   "${CMAKE_C_FLAGS_RELEASE} -D_CRT_SECURE_NO_WARNINGS")
-    set (CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO} -D_CRT_SECURE_NO_WARNINGS")
-    set (CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} -D_CRT_SECURE_NO_WARNINGS")
-    set (CMAKE_CXX_FLAGS_DEBUG   "${CMAKE_CXX_FLAGS_DEBUG} -D_CRT_SECURE_NO_WARNINGS /bigobj")
-    set (CMAKE_C_FLAGS_DEBUG     "${CMAKE_C_FLAGS_DEBUG} -D_CRT_SECURE_NO_WARNINGS /bigobj")
+    # Applies to all configurations
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+    # Avoid: fatal error C1128: number of sections exceeded object file format limit: compile with /bigobj
+    set_source_files_properties(core_validation.cpp threading.cpp
+        PROPERTIES COMPILE_FLAGS "/bigobj")
     # Turn off transitional "changed behavior" warning message for Visual Studio versions prior to 2015.
     # The changed behavior is that constructor initializers are now fixed to clear the struct members.
     add_compile_options("$<$<AND:$<CXX_COMPILER_ID:MSVC>,$<VERSION_LESS:$<CXX_COMPILER_VERSION>,19>>:/wd4351>")

